### PR TITLE
goreleaser for validium node

### PR DESCRIPTION
**Pull request by christophercampbell**
Closes #71. Adds goreleaser to github workflow

### CODE REVIEW

File: Dockerfile.release [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM alpine:3.16.0
+
+EXPOSE 8444
+
+COPY zkevm-node /usr/local/bin
+
+RUN addgroup -S zkevm-group \
+    && adduser -S zkevm-user -G zkevm-group
+
+USER cdk-dac-user
```
Comment (Stefan-Ethernal) [inline Dockerfile.release]: Typo?

```suggestion
USER zkevm-user
```

File: Dockerfile.release [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM alpine:3.16.0
+
+EXPOSE 8444
```
Comment (Stefan-Ethernal) [inline Dockerfile.release]: Should we expose the `8123` port perhaps? 

For the reference: https://github.com/0xPolygon/cdk-validium-node/blob/5f0320c0958e62ffc22b0edfa26a61d0a5dcfc22/Dockerfile#L18

```suggestion
EXPOSE 8123
```

File: .goreleaser.yaml [hunk 3]
```diff
@@ -1,22 +1,78 @@
...
+      - amd64
+      - arm64
+    env:
+      - CGO_ENABLED=0
+    ldflags:
+      - -s -w
+      - -X github.com/0xPolygon/cdk-validium-node.Version={{ .Version }}
+      - -X github.com/0xPolygon/cdk-validium-node.GitRev={{ .Commit }}
+      - -X github.com/0xPolygon/cdk-validium-node.BuildDate={{ .Date }}
+      - -X github.com/0xPolygon/cdk-validium-node.GitBranch={{ .Branch }}
```
Comment (Stefan-Ethernal) [inline .goreleaser.yaml]: Nitpick, though not sure will it do the trick (should be probably tried out):

```suggestion
      - -X github.com/0xPolygon/{{ .ProjectName }}.Version={{ .Version }}
      - -X github.com/0xPolygon/{{ .ProjectName }}.GitRev={{ .Commit }}
      - -X github.com/0xPolygon/{{ .ProjectName }}.BuildDate={{ .Date }}
      - -X github.com/0xPolygon/{{ .ProjectName }}.GitBranch={{ .Branch }}
```

**Review by Stefan-Ethernal — APPROVED** [review]

File: .goreleaser.yaml [hunk 4]
```diff
@@ -1,22 +1,78 @@
...
+      - -X github.com/0xPolygon/{{ .ProjectName }}.GitBranch={{ .Branch }}
+
+archives:
+  - files:
+      - LICENSE
+      - README.md
+
+dockers:
+  - image_templates:
+      - 0xpolygon/{{ .ProjectName }}:{{ .Version }}-amd64
```
Comment (vcastellm) [inline .goreleaser.yaml]: We should tweek the version here as it will try to push `+cdk` tags that are not compatible with docker

**Review by vcastellm — CHANGES_REQUESTED** [review]
Even though the differences are pretty clear, we should try to come up with a way to maintain this in another file than `.goreleaser.yml`

Ideally for me, we should try to change the way docker images are build in zkevm so we can use basically the same file.

File: .goreleaser.yaml (hunk 4 above)
Comment (christophercampbell) [reply .goreleaser.yaml]: @vcastellm would changing the `+cdk` to `-cdk` work? As in

{{ replace .Version "+" "-" }}

Comment (christophercampbell) [reply .goreleaser.yaml]: And do all the uses of Version have to change? name_templates, image_templates, flag_templates?

Comment (vcastellm) [reply .goreleaser.yaml]: hmm only the strings that correspond to docker tags

**Review by vcastellm — APPROVED** [review]
I don't want to be a blocker here

**Pull request merged**